server/services/courseService.ts

```diff
@@ -4,6 +4,22 @@ import type { Course, CourseOffering, CourseParticipation, Person } from '@accre
 export default class CourseService {
   constructor(private readonly courseClientBuilder: RestClientBuilder<CourseClient>) {}
 
+  async createParticipation(
+    token: Express.User['token'],
+    prisonNumber: Person['prisonNumber'],
+    courseId: Course['id'],
+    otherCourseName: string,
+  ): Promise<CourseParticipation> {
+    const courseClient = this.courseClientBuilder(token)
+    const isOtherCourse = courseId === 'other'
+
+    return courseClient.createParticipation(
+      prisonNumber,
+      isOtherCourse ? undefined : courseId,
+      isOtherCourse ? otherCourseName : undefined,
+    )
```

Comment:
Interesting! I think this should be either in the service or the controller rather than the client

Is this tied to the specific way the new participations page works? Would there ever be a case where we need to use this service method with a course that has an `otherCourseName` but without having set `courseId` to `'other'`? Perhaps not

Reply:
I initially had it in the client but then thought it would be nicer in the service.  However now you mention it, 'other' is linked to the functionality of the page template so might make sense to keep anything related to courseId === 'other' in the controller.

Reply:
Yeah, I don't know if we'll ever need this to work with other pages, but it does feel more like a page-specific expectation

Reply:
Moved to controller, see #222